### ISSUE TITLE
[8.0] Timesheet sheet summary widget crash after the creation of a x2many field on hr.analytic.timesheet model

### ISSUE THREAD

**Issue by adrienpeiffer** [odoo/odoo#9279]
Hi,

To find out if an analytical line is a timesheet line, I added an inverse One2Many relation on analytic line like this :
```hr_analytic_timesheet_ids = fields.One2many(
        comodel_name='hr.analytic.timesheet', inverse_name='line_id',
        string='Timesheet Line')```
It works perfectly except when using the summary widget on timesheet sheet.
the create method of analytic line is called with ```{...,'hr_analytic_timesheet_ids': ID, ...``` in vals dict.

So an exception in raised in https://github.com/odoo/odoo/blob/8.0/openerp/osv/fields.py#L798

Here a scenario to reproduce this one on 8.0

https://youtu.be/ue0bdqR5WzY

**Comment by blaggacao** [comment]
Form what I see, I can't judge if this is a bug or inappropriate handling, can you probably specify, why you think this is a bug and should work?

**Comment by adrienpeiffer** [comment]
OPW 680795

**Comment by adrienpeiffer** [comment]
To summarize simply, it's not possible to add x2many fields on hr.analytic.timesheet.

**Comment by blaggacao** [comment]
Ok, you might want to close this issue as part of the shared housekeeping efforts. Best

**Comment by adrienpeiffer** [comment]
@blaggacao What do you mean exactly ?

**Comment by blaggacao** [comment]
As the reporter of the issue, you can close it here directly in github by clicking on the close button. If I understand you correctly, it is an expected/known behaviour. This means that without further and detailed reasoning, why this is still not correct behaviour, one should be generally in favor of closing issues.

**Comment by adrienpeiffer** [comment]
Currently, if you add an x2many field on hr.analytic.line the javascript widget for timesheet encoding causes crash. This is not an expected behavior. (See the crash at the end of the video that you can find just above). I don't really understand why I should close this one.

**Comment by beledouxdenis** [comment]
Dear Adrien,

We do not consider this issue to be solved in Odoo 8.0, because this specific widget was just not designed for the use you want from it.

This widget just does not support *2many fields, and we won't add the support in as stable release such as Odoo 8.0, as this could obviously lead to new bugs, in a stable release, which we do our best to avoid.

Besides, it doesn't lead to any issue in Odoo standard.
If you need this support for one of your custom module, just add this support by extending the according widget in your custom module.

I hope you will understand.

Regards,

**Comment by adrienpeiffer** [comment]
Hi Denis,

Thanks for your answer.

> We do not consider this issue to be solved in Odoo 8.0, because this specific widget was just not designed for the use you want from it.

The usage the widget is not related to the functionality that I want added. I just want to add a x2many field on the hr.analytic.line model without considering the widget.

And, when using this widget (Without using the new field), there is a crash.

To summarize simply, this mean that it is not possible to add a x2many field on hr.analytic.timesheet model (Regardless of the widget).

So you're not going to fix this?

Thanks in advance,

Adrien
